charts/zitadel/templates/setupjob.yaml

```diff
@@ -157,7 +164,87 @@ spec:
+          command:
+            - sh
+            - -c
+            - |
+              #!/bin/sh
+              set -e
+
+              NAMESPACE="{{ .Release.Namespace }}"
+              SECRET_NAME="{{ .Values.zitadel.configmapConfig.FirstInstance.Org.Machine.Machine.Username }}"
+              KEY_FILE_PATH="/machinekey/sa.json"
+              MAIN_CONTAINER_NAME="{{ .Chart.Name }}-setup"
+
+              echo "Waiting for main container '${MAIN_CONTAINER_NAME}' to terminate..."
+
+              while [ -z "$(kubectl --namespace="${NAMESPACE}" get pod "${POD_NAME}" --output=jsonpath="{.status.containerStatuses[?(@.name=='${MAIN_CONTAINER_NAME}')].state.terminated}")" ]; do
+                sleep 5
+              done
+
+              echo "Main container has terminated."
+
+              if [ -f "${KEY_FILE_PATH}" ]; then
+                echo "File found at '${KEY_FILE_PATH}'. Creating secret..."
+                kubectl --namespace="${NAMESPACE}" \
+                  create secret generic "${SECRET_NAME}" \
+                  --from-file="${SECRET_NAME}.json=${KEY_FILE_PATH}"
+                echo "Secret '${SECRET_NAME}' created successfully."
+              else
+                echo "File NOT found at '${KEY_FILE_PATH}'. Setup cannot continue."
+                exit 1
+              fi
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+          volumeMounts:
+            - name: machinekey
+              mountPath: "/machinekey"
+              readOnly: true
+          resources:
+          {{- if .Values.setupJob.machinekeyWriter.resources }}
+            {{- toYaml .Values.setupJob.machinekeyWriter.resources | nindent 12 }}
+          {{- else }}
+            {{- toYaml .Values.setupJob.resources | nindent 12 }}
+          {{- end }}
+      {{- end }}
+      {{- if and (not .Values.zitadel.configmapConfig.FirstInstance.Skip) (include "deepCheck" (dict "root" .Values "path" (splitList "." "zitadel.configmapConfig.FirstInstance.Org.Machine.Pat"))) }}
+        - name: "{{ .Chart.Name }}-machine-pat"
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 14 }}
+          image: "{{ .Values.setupJob.machinekeyWriter.image.repository }}:{{ .Values.setupJob.machinekeyWriter.image.tag | default (.Capabilities.KubeVersion.GitVersion | trimPrefix "v") }}"
+          command:
+            - sh
+            - -c
+            - |
+              #!/bin/sh
+              set -e
+
+              NAMESPACE="{{ .Release.Namespace }}"
+              SECRET_NAME="{{ .Values.zitadel.configmapConfig.FirstInstance.Org.Machine.Machine.Username }}-pat"
```

Comment:
I think I remember that that there is an issue with installing the chart multiple times in the same namespace because the secret name already exists on the second installation. Should we pre- or suffix the secret with the release name or so?

Reply:
Maybe also to other secrets we create. This might be a breaking change, though? wdyt?

Reply:
I think it's related to this https://github.com/zitadel/zitadel-charts/pull/429